src/ragas/adaptation.py

```diff
@@ -0,0 +1,33 @@
+import typing as t
+
+from langchain_core.language_models import BaseLanguageModel
+
+from ragas.llms import llm_factory
+from ragas.llms.base import BaseRagasLLM, LangchainLLMWrapper
+from ragas.metrics.base import MetricWithLLM
+
+
+def adapt(
+    metrics: t.List[MetricWithLLM],
+    language: str,
+    llm: t.Optional[BaseRagasLLM] = None,
+    cache_dir: t.Optional[str] = None,
+) -> None:
+    """
+    Adapt the metric to a different language.
+    """
+
+    llm_wraper = None
+
+    if llm is None:
+        llm_wraper = llm_factory()
+    elif isinstance(llm, BaseLanguageModel):
+        llm_wraper = LangchainLLMWrapper(llm)
+
+    for metric in metrics:
+        if metric.llm is None or llm is not None:
+            metric.llm = llm_wraper
```

Comment:
I don't know if this if statement is correct.

if metric has an LLM and the user passes an LLM with adapt functionality then the LLM the user passed metric.llm get replaced right? is that the expected behaviour

Reply:
Yes. Say user has used `gpt-3.5` for eval but need to use `gpt-4` for adapt.